src/ripple/rpc/impl/RPCHelpers.h

```diff
@@ -241,7 +241,7 @@ extern beast::SemanticVersion const lastVersion;
 constexpr unsigned int apiInvalidVersion = 0;
 constexpr unsigned int apiVersionIfUnspecified = 1;
 constexpr unsigned int apiMinimumSupportedVersion = 1;
-constexpr unsigned int apiMaximumSupportedVersion = 1;
+constexpr unsigned int apiMaximumSupportedVersion = 2;
 constexpr unsigned int apiBetaVersion = 2;
```

Comment:
As I wrote over [on #4571](https://github.com/XRPLF/rippled/pull/4571#discussion_r1228663775), once a version of `rippled` is released with `apiMaximumSupportedVersion` is set to 2, then we are saying that API v2 is **Done**. Once that happens we can not legitimately make any further changes to API version 2. To the best of my knowledge, API version 2 is still in beta, under active development, and still changing. 

Now, if you (and #4571 and #4552) are saying that version 2 is done, let's talk about version 2 being done. This will be the first I've heard about it.

If you want a node to properly handle API v2 requests, the correct way to do that, for now, is with the `[beta_rpc_api]` config option. And if you want to do it in unit tests, set it up in the config similar to how it's done in https://github.com/XRPLF/rippled/blob/develop/src/test/rpc/AccountInfo_test.cpp#L209-L212, or just by setting `cfg.BETA_RPC_API=true;` directly.

Reply:
Please see also #4573.

Reply:
Let's say in 1.12 that API v2 is done. Any changes to the API after that point will be for api_version 3+.

Reply:
It seems in that case, we might need to work (or get clarity) on the versioning scheme a bit more and get answers to the questions that @thejohnfreeman posted [here](https://github.com/XRPLF/rippled/pull/4552#discussion_r1230276163)